bytecomp/translprim.ml

```diff
@@ -692,10 +696,26 @@ let lambda_of_prim prim_name prim loc args arg_exps =
     | Send | Send_self | Send_cache), _ ->
       raise(Error(loc, Wrong_arity_builtin_primitive prim_name))
 
+let check_primitive_arity loc p =
+  let prim = lookup_primitive loc p in
+  let ok =
+    match prim with
+    | Primitive (_,arity) -> arity = p.prim_arity
+    | External _ -> true
+    | Comparison _ -> p.prim_arity = 2
+    | Raise _ -> p.prim_arity = 1
+    | Raise_with_backtrace -> p.prim_arity = 2
+    | Lazy_force -> p.prim_arity = 1
+    | Loc _ -> p.prim_arity = 1 || p.prim_arity = 0
+    | Send | Send_self -> p.prim_arity = 2
+    | Send_cache -> p.prim_arity = 4
```

Comment:
What do you think about writing this as a pattern matching on `(prim, p.prim_arity)` ?

Reply:
I didn't do it because the last pattern would have to either catch all or enumerate all constructors.

Reply:
OK, fair enough.